docker-compose.yaml

```diff
@@ -83,6 +95,10 @@ services:
       - SPARK_MASTER_URL=spark://spark-master:7077
       - REDIS_HOST=redis
       - REDIS_PORT=6379
+      - SPARK_EVENT_LOG_DIR=s3a://cdm-spark-events/spark-events
```

Comment:
is it ok for a bunch of different spark instances to log to the same place? 

Do we need to worry about log rotation / deletion?

Reply:
I guess you could add a log ID env var and put the logs in a subdirectory based on that env var

Reply:
The issue with that approach is that the Spark job won't be able to create the subdirectory in S3. The only workaround I can think of is modifying the setup script to create the subdirectory ahead of time - but that would expose the MinIO admin credentials to all Spark nodes.

```
py4j.protocol.Py4JJavaError: An error occurred while calling None.org.apache.spark.api.java.JavaSparkContext.
: java.io.FileNotFoundException: No such file or directory: s3a://cdm-spark-job-logs/spark-job-logs/user-notebook
```

Reply:
That's wild, considering that directories don't actually exist in s3

Reply:
> modifying the setup script to create the subdirectory ahead of time - but that would expose the MinIO admin credentials to all Spark nodes.

Hold on - why would the setup script require admin creds? It already has to have write creds which means it should be able to "create" a directory